mysticeti-core/src/commit_observer.rs

```diff
@@ -246,6 +249,20 @@ impl SimpleCommitObserver {
             }
         }
     }
+
+    fn report_metrics(&self, committed: &[CommittedSubDag]) {
+        let utc_now = timestamp_utc();
+        let mut total = 0;
+        for block in committed.iter().flat_map(|dag| &dag.blocks) {
+            let latency_ms = utc_now
+                .checked_sub(block.meta_creation_time())
+                .unwrap_or_default();
+
+            total += 1;
+            self.metrics.block_commit_latency.observe(latency_ms);
+        }
+        self.metrics.blocks_per_commit_count.observe(total);
```

Comment:
I think interesting metric can also be just number of SubDags per this call - w/o multi leader it should be ~1 in normal case(more then 1 subdag committed at a time mean we fallback to recursive commit rule)